src/library/Connect/LocalEnvironmentManager.cs

```diff
@@ -436,12 +436,26 @@ private void _ReportProgress(EventLevel eventLevel, string message, params objec
 
                 // because we are using dns name instead of service we have to retrieve it by splitting when needed
                 // If this ever cause issues we should consider larger refactor where we add serviceName member variable to EndpointInfo class.
-                var serviceName = endpoint.DnsName
+                var dnsNameArray = endpoint.DnsName
                     .ToUpperInvariant()
-                    .Split(".")
+                    .Split(".");
+
+                var serviceName = dnsNameArray
                     .First()
-                    .Replace("-", "_")
-                    .Replace(".", "_");
+                    .Replace("-", "_");
+
+                // when !endpoint.IsInWorkloadNamespace
+                var serviceNs = dnsNameArray.Length == 2
+                    ? dnsNameArray
+                        .Last()
+                        .Replace("-", "_")
+                    : null;
+
+                // sometimes cross talk is desired between namespaces, append those not matching workload
+                if (!string.IsNullOrWhiteSpace(serviceNs))

```

Comment:
right now we can implicitly infer this when `!endpoint.IsInWorkloadNamespace`, as the namespace is [conditionally applied](https://github.com/Azure/Bridge-To-Kubernetes/blob/51ea890699b14a3213045b7e72625aea9db170b2/src/library/Connect/WorkloadInformationProvider.cs#L466)